blockly-ws-search/src/WorkspaceSearch.js

```diff
@@ -0,0 +1,660 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Object responsible for workspace search.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ * @author kozbial@google.com (Monica Kozbial)
+ */
+
+
+import { injectSearchCss } from './css.js';
+import * as Blockly from 'blockly/core';
+
+export class WorkspaceSearch {
+  /**
+   * Class for workspace search.
+   * @param {!Blockly.WorkspaceSvg} workspace
+   */
+  constructor(workspace) {
+    /**
+     * The workspace the search bar sits in.
+     * @type {!Blockly.WorkspaceSvg}
+     * @private
+     */
+    this.workspace_ = workspace;
+
+    /**
+     * HTML container for the search bar.
+     * @type {?HTMLElement}
+     * @private
+     */
+    this.htmlDiv_ = null;
+
+    /**
+     * The div that holds the search bar actions.
+     * @type {?HTMLElement}
+     * @protected
+     */
+    this.actionDiv_ = null;
+
+    /**
+     * The text input for the search bar.
+     * @type {?HTMLInputElement}
+     * @private
+     */
+    this.inputElement_ = null;
+
+    /**
+     * The placeholder text for the search bar input.
+     * @type {string}
+     * @private
+     */
+    this.textInputPlaceholder_ = 'Search';
+
+    /**
+     * A list of blocks that came up in the search
+     * @type {!Array.<Blockly.BlockSvg>}
+     * @protected
+     */
+    this.blocks_ = [];
+
+    /**
+     * Index of the currently "selected" block in the blocks array.
+     * @type {number}
+     * @protected
+     */
+    this.currentBlockIndex_ = -1;
+
+    /**
+     * Currently "selected" block.
+     * @type {Blockly.BlockSvg}
+     * @protected
+     */
+    this.currentBlock_ = null;
```

Comment:
I don't see uses for this where ``this.block_[this.currentBlockIndex_]]`` can't be used.

Reply:
It should be fine to replace `this.currentBlock_` with `this.block_[this.currentBlockIndex_]`